File: backend/RentLoop/RentLoop.API/DTOs/Reviews/ReviewCreateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace RentLoop.API.DTOs.Reviews
{
    public class ReviewCreateRequest
    {
        public int ReservationId { get; set; }

        [Range(1, 5)]
        public int Rating { get; set; }

        [MaxLength(1000)]
        public string? Comment { get; set; }
    }
}

File: frontend/rentloop_app/lib/features/user/models/review_create_request.dart
class ReviewCreateRequest {
  final int reservationId;
  final int rating; // 1..5
  final String comment;

  ReviewCreateRequest({
    required this.reservationId,
    required this.rating,
    required this.comment,
  });

  Map<String, dynamic> toJson() => {
        'reservationId': reservationId,
        'rating': rating,
        'comment': comment,
      };
}
